relay-metrics/src/aggregation.rs

```diff
@@ -69,6 +71,403 @@ impl GaugeValue {
     }
 }
 
+/// A distribution of values within a [`Bucket`].
+///
+/// Distributions store a histogram of values. It allows to iterate both the distribution with
+/// [`iter`](Self::iter) and individual values with [`iter_values`](Self::iter_values).
+///
+/// Based on individual reported values, distributions allow to query the maximum, minimum, or
+/// average of the reported values, as well as statistical quantiles.
+///
+/// # Example
+///
+/// ```rust
+/// use relay_metrics::dist;
+///
+/// let mut dist = dist![1.0, 1.0, 1.0, 2.0];
+/// dist.insert(5.0);
+/// dist.insert_multi(3.0, 7);
+/// ```
+///
+/// Logically, this distribution is equivalent to this visualization:
+///
+/// ```plain
+/// value | count
+/// 1.0   | ***
+/// 2.0   | *
+/// 3.0   | *******
+/// 4.0   |
+/// 5.0   | *
+/// ```
+///
+/// # Serialization
+///
+/// Distributions serialize as sorted lists of floating point values. The list contains one entry
+/// for each value in the distribution, including duplicates.
+#[derive(Clone, Default, PartialEq)]
+pub struct DistributionValue {
+    singles: BTreeSet<FloatOrd<f64>>,
+    duplicates: BTreeMap<FloatOrd<f64>, usize>,
```

Comment:
do we ever want to use platform-specific integer sizes here?

Reply:
You're right. I chose `usize` since I thought it doesn't matter and since we can't handle more than usize individual values in one bucket anyway. Will change to `u64` or maybe even `u32` to clarify intent, however.